### PR TITLE
feat(TCOMP-507): Unable to get a UI Spec with existing data

### PR DESCRIPTION
**Pull request by charlesnguyen**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TCOMP-507

**What is the new behavior?**
A new end point has been added : 
POST /properties/{definitionName}?formName=MAIN

**Please check if the PR fulfills these requirements**

- [ x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java [hunk 1]
```diff
@@ -190,6 +190,12 @@ public String getProperties(String definitionName, String formName) {
         return new ResponseEntity<>(response, OK);
     }
 
+    public String initializeProperties(String formName, PropertiesDto propertiesContainer) {
+        Properties properties = propertiesHelpers.propertiesFromDto(propertiesContainer);
+        properties.refreshLayout(properties.getForm(formName));
+        return properties == null ? "{}" : jsonSerializationHelper.toJson(formName, properties);
```
Comment (sgandon) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java]: if the `properties` instance may be null then you have to make sure the code in the line above does not cause an NPE and check for null before calling the *refreshLayout*

File: services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Install all of the definitions provided for the FullExample family of components.
+ */
+@AutoService(ComponentInstaller.class)
+@Component(name = Constants.COMPONENT_INSTALLER_PREFIX + FullExampleFamilyDefinition.NAME, provide = ComponentInstaller.class)
+public class FullExampleFamilyDefinition extends AbstractComponentFamilyDefinition implements ComponentInstaller {
+
+    public static final String NAME = "FullExample";
+
+    public FullExampleFamilyDefinition() {
+        super(NAME, new org.talend.components.service.rest.fullexample.FullExampleInputDefinition(), new FullExampleDatastoreDefinition(), new FullExampleDatasetDefinition());
```
Comment (sgandon) [inline services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java]: You don't need the FullExampleInputDefinition for your test so I suggest you remove it and remove the class too (FullExampleInputDefinition.java).

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/454/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/454/)

File: services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java (hunk 2 above)
Comment (sgandon) [inline services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java]: And do you need a the Datastore too cause it seems that the test of initialising properly the ui-specs is done on the dataset ?

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java (hunk 1 above)
Comment (charlesnguyen) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java]: great point :)

File: services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java (hunk 2 above)
Comment (charlesnguyen) [inline services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleFamilyDefinition.java]: I test both Datastore  and Dataset.

**Comment by gonchar-ivan** [comment]
As I understand this PR adds new endpoint
POST /properties/{definitionName}
What about documentation for it?
I found we have 
https://github.com/Talend/components/wiki/7.-Components-REST-service#validate-component-properties

but new method is rather _initialize_ method then validate

**Pull request merged**